crates/cast/bin/cmd/send.rs

```diff
@@ -87,103 +91,235 @@ pub enum SendTxSubcommands {
     },
 }
 
+#[derive(Debug, Parser)]
+#[command(next_help_heading = "Bump gas price options")]
+struct BumpGasPriceArgs {
+    /// Enable automatic gas price escalation for transactions.
+    ///
+    /// When set to true, automatically increase the gas price of a pending transaction. It can be
+    /// used to replace transactions that are stuck during busy network times.
+    #[arg(long, alias = "bump-fee")]
+    auto_bump_gas_price: bool,
+
+    // The percentage by which to increase the gas price on each retry.
+    #[arg(long, default_value = "10")]
+    gas_price_increment_percentage: u64,
+
+    /// The maximum total percentage increase allowed for gas price.
+    ///
+    /// This sets an upper limit on the gas price across all retry attempts, expressed as a
+    /// percentage of the original price. For example, a value of 150 means the gas price will
+    /// never exceed 150% of the original price (1.5 times the initial price).
+    #[arg(long, default_value = "150")]
+    gas_price_bump_limit_percentage: u64,
+
+    /// The maximum number of times to bump the gas price for a transaction.
+    #[arg(long, default_value = "3")]
+    max_gas_price_bumps: u64,
+}
+
 impl SendTxArgs {
     #[allow(unknown_lints, dependency_on_unit_never_type_fallback)]
     pub async fn run(self) -> Result<(), eyre::Report> {
```

Comment:
I think we should make it a no-op if `get_nonce(from, "latest") == get_nonce(from "pending")` to avoid footguns. and also make sure to set `tx.nonce` to avoid it being filled automatically

Reply:
also wondering if this should conflict with `--async` or disable it automatically?

Reply:
I think I know how to get `get_nonce(from, "latest")`:

```rust
let sender = eth.wallet.from.unwrap();
let nonce = provider.get_transaction_count(sender);
```

But how do I get `get_nonce(from "pending")`?

Reply:
Regarding your second comment, I think `--bump-fee` conflicts with `--async`. I'll just add `conflicts_with = "async"`.

Reply:
to get pending nonce you can do `provider.get_transaction_count(from).block_id(BlockId::pending())`

> edited: Regarding your second comment, I think `--bump-fee` and `--async` can still be used together? We can just replace a pending tx by another pending tx with increased gas price?

right, this should be fine

Reply:
Addressed by [e01ecf8](https://github.com/foundry-rs/foundry/pull/9147/commits/e01ecf851e491d351a313665e54d225616f28dd0)